tests/Unit/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/Test_GaugeWave.cpp

```diff
@@ -0,0 +1,234 @@
+  const auto one = make_with_value<DataType>(x, 1.);
+  for (size_t i = 0; i < 3; ++i) {
+    CHECK(shift.get(i) == zero);
+    CHECK(dt_shift.get(i) == zero);
+    if (i > 0) {
+      CHECK(d_lapse.get(i) == zero);
+    }
+    for (size_t j = 0; j < 3; ++j) {
+      CHECK(d_shift.get(i, j) == zero);
+      if (i != j) {
+        CHECK(g.get(i, j) == zero);
+      } else {
+        if (i > 0) {
+          CHECK(g.get(i, j) == one);
+        }
+      }
+      if (i > 0 or j > 0) {
+        CHECK(dt_g.get(i, j) == zero);
+        for (size_t k = 0; k < 3; ++k) {
+          CHECK(d_g.get(k, i, j) == zero);
+        }
+      } else {
+        for (size_t k = 1; k < 3; ++k) {
+          CHECK(d_g.get(k, i, j) == zero);
+        }
+      }
+    }
+  }
+
+  const auto& h =
+      1.0 - amplitude * sin((2.0 * M_PI / wavelength) * (get<0>(x) - t));
+  const auto& dx_h = -1.0 * (2.0 * M_PI / wavelength) * amplitude *
+                     cos((2.0 * M_PI / wavelength) * (get<0>(x) - t));
+  const auto& sqrt_h = sqrt(h);
+  const auto& dx_h_over_2_sqrt_h = 0.5 * dx_h / sqrt_h;
+
+  // Check quantities that depend on h
+  CHECK(get(lapse) == sqrt_h);
+  CHECK(get(dt_lapse) == -1.0 * dx_h_over_2_sqrt_h);
+  CHECK(get<0>(d_lapse) == dx_h_over_2_sqrt_h);
+  CHECK(get<0, 0>(g) == h);
+  CHECK(get<0, 0>(dt_g) == -1.0 * dx_h);
+  CHECK(get<0, 0, 0>(d_g) == dx_h);
+
+  // Check quantities derivable from spatial metric, lapse, shift, and their
+  // derivatives
+  const auto& invg =
+      get<gr::Tags::InverseSpatialMetric<3, Frame::Inertial, DataType>>(vars);
+  const auto& sqrt_detg = get<gr::Tags::SqrtDetSpatialMetric<DataType>>(vars);
+  const auto& det_and_inverse = determinant_and_inverse(g);
+  const auto& expected_invg = det_and_inverse.second;
+  const auto& expected_sqrt_detg = sqrt(get(det_and_inverse.first));
+  CHECK_ITERABLE_APPROX(invg, expected_invg);
+  CHECK_ITERABLE_APPROX(get(sqrt_detg), expected_sqrt_detg);
+
+  const auto& K =
+      get<gr::Tags::ExtrinsicCurvature<3, Frame::Inertial, DataType>>(vars);
+  const auto& expected_K =
+      gr::extrinsic_curvature(lapse, shift, d_shift, g, dt_g, d_g);
+  CHECK_ITERABLE_APPROX(K, expected_K);
+}
+
+void test_serialize() noexcept {
+  gr::Solutions::GaugeWave solution(3.0, 4.0);
+  test_serialization(solution);
+}
+
+void test_copy_and_move() noexcept {
+  gr::Solutions::GaugeWave solution(3.0, 4.0);
+  test_copy_semantics(solution);
+  auto solution_copy = solution;
+  // clang-tidy: std::move of trivially copyable type
+  test_move_semantics(std::move(solution), solution_copy);  // NOLINT
+}
+
+void test_construct_from_options() {
+  Options<tmpl::list<GaugeWave>> opts("");
+  opts.parse(
+      "GaugeWave:\n"
+      "  Amplitude: 0.5\n"
+      "  Wavelength: 4.0");
+  CHECK(opts.get<GaugeWave>() == gr::Solutions::GaugeWave(0.5, 4.0));
+}
+
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.PointwiseFunctions.AnalyticSolutions.Gr.GaugeWave",
+                  "[PointwiseFunctions][Unit]") {
```

Comment:
You are currently testing this solution by evaluating it with a set of literal numbers. I think ideally we'd like to have a test with literal numbers (to be sure the solution is correct at least for those numbers) and one that compares to a Python implementation with random numbers (to turn up any edge cases). Would you be ok with adding a Python test for this? It's a very simple solution, but since people copy code and tests around a lot we should probably set good examples :)

Reply:
12/16: add python test as well, so code that generated the known answer is also in the repo